hibernate-core/src/main/java/org/hibernate/query/sqm/sql/internal/EntityValuedPathInterpretation.java

```diff
@@ -318,7 +319,11 @@ public static <T> EntityValuedPathInterpretation<T> from(
 			if ( discriminatorMapping != null ) {
 				discriminatorMapping.forEachSelectable( selectableConsumer );
 			}
-			entityMappingType.forEachSelectable( selectableConsumer );
+			entityMappingType.visitFetchables( fetchable -> {
```

Comment:
Minor nitpick. Could you please use a loop based on `getFetchable(int)` and `getNumberOfFetchables()`

Reply:
Done!